File: graph/src/main/groovy/net/twisterrob/gradle/graph/vis/LoggingTaskVisualizer.java
package net.twisterrob.gradle.graph.vis;

import java.util.Map;

import org.gradle.api.*;
import org.gradle.api.initialization.Settings;

import net.twisterrob.gradle.graph.tasks.*;

public class LoggingTaskVisualizer implements TaskVisualizer {
	private final TaskVisualizer wrapped;

	LoggingTaskVisualizer(TaskVisualizer wrapped) {
		this.wrapped = wrapped;
	}

	@Override public void showUI(Settings project) {
		System.out.printf("showUI(%s)\n", project);
		wrapped.showUI(project);
	}
	@Override public void initModel(Map<Task, TaskData> graph) {
		System.out.printf("initModel(%s)\n", graph);
		wrapped.initModel(graph);
	}
	@Override public void update(Task task, TaskResult result) {
		System.out.printf("update(%s, %s)\n", task, result);
		wrapped.update(task, result);
	}
	@Override public void closeUI() {
		System.out.printf("closeUI()\n");
		wrapped.closeUI();
	}
}

File: graph/src/main/kotlin/net/twisterrob/gradle/graph/vis/LoggingTaskVisualizer.kt
package net.twisterrob.gradle.graph.vis

import net.twisterrob.gradle.graph.tasks.TaskData
import net.twisterrob.gradle.graph.tasks.TaskResult
import org.gradle.api.Task
import org.gradle.api.initialization.Settings
import org.slf4j.Logger

class LoggingTaskVisualizer(
	private val wrapped: TaskVisualizer,
	private val logger: Logger,
) : TaskVisualizer {

	override fun showUI(settings: Settings) {
		logger.debug("showUI({})", settings)
		wrapped.showUI(settings)
	}

	override fun initModel(graph: Map<Task, TaskData>) {
		logger.debug("initModel({})", graph)
		wrapped.initModel(graph)
	}

	override fun update(task: Task, result: TaskResult) {
		logger.debug("update({}, {})", task, result)
		wrapped.update(task, result)
	}

	override fun closeUI() {
		logger.debug("closeUI()")
		wrapped.closeUI()
	}
}
